muspinsim/input/structure.py

```diff
@@ -287,21 +287,27 @@ def compute_closest(
         Raises:
             RuntimeError -- When the expansion tries to go beyond the maximum
                             limit given by max_layer.
+            RuntimeError -- When no atoms are found
         """
 
         # Current list of atoms in expanded supercell
         atoms = self._cell_atoms
         if ignored_symbols:
             atoms = list(filter(lambda atom: atom.symbol not in ignored_symbols, atoms))
 
+        if len(atoms) == 0:
```

Comment:
Just trying this out, we always require 1 muon in the structure, by default represented by H. If I ignore H along with the other atoms (like the test does), then we have a length of 0 and the error is raised as intended. But if I ignore everything _except_ the muon (H), then I'll pass this test because the length is 1, but when I expand the cell we don't include the muon in the supercell, and `new_atoms` still ends up as an empty list.

Another problem I've just spotted, passing `--ignore_symbol H` (or more generally, ignoring whatever the muon symbol is) will cause trouble since `atoms = atoms[1 : (number + 1)]` further down implicitly relies on the first atom being the muon (but if we ignore it, it won't be in `atoms` at all - resulting in us ignoring the actual closest atom which ends up with index 0). I think the simplest thing here is to raise an error if the same symbol is both ignored and used to identify the muon (maybe this will have to go higher up in `_run_generator_tool`?) since ignoring the symbol of the muon doesn't really make sense?

If we add the check above I think we can then just check if the length is `<=1`, since we require at least one atom in addition to the muon (which should now always be in the list).

```suggestion
        if len(atoms) == 1:
```

Reply:
Thanks for spotting these, I think that's a good approach. I don't see any reason to allow the muon to be ignored like that.